### ISSUE TITLE
Pandoc command doesn't work in vim with python3 support

### ISSUE THREAD

**Issue by wuy3814** [vim-pandoc/vim-pandoc#81]
The formatting is perfect but when it comes to compile the markdown script, it failed saying "E464: Ambiguous use of user-defined command ".
The problem seemed to be the python support, since  :echo v:version >= 704 && has("python") always returns "0".  
However, I already installed Python 3.1.

**Comment by lars-becker** [comment]
I have the same problem, although with a different vim build (that which is shipped with the current OpenSUSE distribution, 7.4.461 with `+python/dyn` and `+python3/dyn`). My build is shipped *with support for Python 2 and Python 3*. Since one other plugin uses Python 3 only python3 is available. This leads to the above mentioned error message and the `:Pandoc` command is not available.

**Comment by fmoralesc** [comment]
@lars-becker I just pushed a fix for that scenario, please confirm if the issue is fixed.

**Comment by lars-becker** [comment]
Hi @fmoralesc, I tried it with your changes, but that doesn't work either. Vim throws an `E464: "Ambiguous use of user-defined command"`.

**Comment by SuperFluffy** [comment]
I also get the `E464: Ambiguous use of user-defined command` error. It's seems like the `:Pandoc` command is not available. My `vim` build comes with Python. `:echo has('python3')` reports `1`.

**Comment by fmoralesc** [comment]
@SuperFluffy that is a proper bug, I hadn't tested it against a vim with only python3.

**Comment by fmoralesc** [comment]
I just pushed support for python3 to master, please test.

**Comment by SuperFluffy** [comment]
I can run `:Pandoc` on my `vim` now. Howevere, it seems to choke on long options. Compare:
```
:Pandoc html -s -S -N
> vim-pandoc: ran successfully
```

And
```
:Pandoc html -s -S --numbers-sections
> Error detected while processing function pandoc#command#Pandoc:
line    5:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/$USER/.vim/bundle/vim-pandoc/pythonx/vim_pandoc/command.py", line 204, in __call__
    c_opts, c_args = getopt.gnu_getopt(shlex.split(args), self.opts.shortopts, self.opts.longopts)
  File "/usr/lib/python3.4/getopt.py", line 136, in gnu_getopt
    opts, args = do_longs(opts, args[0][2:], longopts, args[1:])
  File "/usr/lib/python3.4/getopt.py", line 157, in do_longs
    has_arg, opt = long_has_args(opt, longopts)
  File "/usr/lib/python3.4/getopt.py", line 174, in long_has_args
    raise GetoptError(_('option --%s not recognized') % opt, opt)
getopt.GetoptError: option --number-sections not recognized
```

This is especially a problem for options for which only long (i.e. double-dashed) versions exists, e.g. `--toc, --table-of-contents`.

**Comment by fmoralesc** [comment]
Can't reproduce here, although from the cmdline:
~~~ vim
:Pandoc html -s -S --numbers-sections
~~~
`--numbers-sections` doesn't exist in pandoc. 

Please tell me what are your OS and pandoc version, the help parser might be choking on the output of `pandoc --help`, which vim-pandoc uses to generate the list of arguments.

**Comment by SuperFluffy** [comment]
My bad, typo in my comment. The correct flag is `--number-sections` (which I used, as you can see from the error message). The main point is that it seems like long options, i.e. those preceded by two dashes in the form `--<option_name>`, do not seem to work. Please especially check `--toc` for that.

The OS is Arch Linux with `pandoc` release `1.13.2.1`.

**Comment by fmoralesc** [comment]
I can confirm now, will see what I can do.

**Comment by fmoralesc** [comment]
It was a problem with the long options being passed as a `map` instead of a `list`, and the garbage collector reaping them during the lifetime of the command. Should be fixed now, please confirm.

**Comment by SuperFluffy** [comment]
It works now. `:Pandoc html -s -S --number-sections --toc` runs beautifully and produces exactly what it should.

**Comment by fmoralesc** [comment]
:)

**Comment by daaronr** [comment]
https://github.com/vim-pandoc/vim-pandoc/issues/81#issuecomment-106767983
> I just pushed support for python3 to master, please test.

 @fmoralesc

I'm having the same issue now that @SuperFluffy had, three years later. All was working until I installed Mojave (and simultaneously, installed markdown support for R/Rstudio).
Now, when I try to compile using the Pandoc tab (or with the command) I get the "E464: Ambiguous use" message.

I've reinstalled the latest version of Pandoc and Python. 
Any advice would be greatly appreciated.
